zeppelin-server/src/main/java/org/apache/zeppelin/service/NotebookService.java

```diff
@@ -1553,4 +1554,21 @@ private <T> boolean checkPermission(String noteId,
       return false;
     }
   }
+
+  private String decodeRepeatedly(final String encoded) throws IOException {
```

Comment:
This method can be static.
```suggestion
  private static String decodeRepeatedly(final String encoded) throws IOException {
```